### DISCUSSION TITLE
Optimization Suggestion

### DISCUSSION THREAD

**Discussion by ghost** [instantOS/instantOS#126] [Ideas]
Regarding lines 16-20 of 'monitor.sh', I recommend converting them to something along these lines, which is a _considerable_ optimization, taking it from 10 program executions to just 1:

```bash
while read -a Line; do
	[ "${Line[1]}" == 'connected' ] || continue
	if [[ ${Line[2]} =~ ^([0-9]+x[0-9]+)\+([0-9]+)\+[0-9]+$ ]]; then
		RESOLUTIONS+=("${BASH_REMATCH[1]}")
		POSITIONS+=("${BASH_REMATCH[2]}")
	fi
done <<< "$(xrandr)"
AMOUNT=${#POSITIONS[@]}
```

The above code is more of a proof of concept. I would have submitted a PR with this improvement, but I'm not entirely sure what you're wanting. Hope this at least sparks some thought.

**Comment by con-f-use** [comment]
You're welcome to open a pull request (PR) so that change can be properly attributed to you and we can discuss more mature versions there and suggest code-changes. That can even be done using github's website directly, where you can also link [ranges of lines](https://github.com/instantOS/instantOS/blob/main/monitor.sh#L16-L20) btw.

![image](https://user-images.githubusercontent.com/11145016/145681606-ed6aa792-e606-4659-8733-ce1d3359b59e.png)

**Reply by ghost** [reply]
Thanks for the reply. Ordinarily, I'd ... well, it said it here: _"I would have submitted a PR with this improvement, but I'm not entirely sure what you're wanting. Hope this at least sparks some thought."_ So, it just became food for thought. I forked this repository quite a while back, and have recently made a change or two, so you may yet get a PR from me. :)

Regarding the code here: it's taking advantage of the ability in modern BASH to capture a REGEX group (the parentheses), thanks to the `BASH_REMATCH` array variable, and it's a single pass through the output of xrandr(1), instead of running it twice to parse the same output. It also uses arrays for the list of resolutions and positions, as that seemed to be the most fitting, albeit not _necessary_.

**Comment by con-f-use** [comment]
>  I would have submitted a PR with this improvement

Yeah, I totally read over that sentence. Nonetheless a PR would be very welcome. Everything else I mentioned here: https://github.com/instantOS/instantSETTINGS/pull/33 tl;dr: Optimizations are welcome and that one is good, but as long as they are not in really performance critical area, they are often more effort than they are worth.
